lib/parallel.rb

```diff
@@ -56,7 +56,11 @@ def work(data)
       end
 
       begin
-        Marshal.load(read)
+        result = Marshal.load(read)
+        if ExceptionWrapper === result
+          raise result.exception
+        end
+        result
       rescue EOFError
```

Comment:
this would make any EOFError in a block turn into DeadWorker, the original code was more like:

```Ruby
result = begin
  Marshal.load(read)
rescue EOFError
  raise DeadWorker
end
raise result.exception if ExceptionWrapper === result
result
```